### ISSUE TITLE
Low level API to generate candidate tokens and probabilities

### ISSUE THREAD

**Issue by m1chae1bx** [marella/ctransformers#15]
The sample() function only samples a token from the current state model. It would be nice to have a low level API to generate candidate tokens with probabilities to make it possible to implement custom deciding strategies.

**Comment by marella** [comment]
Thanks. I added a low level API to access and modify the logits in the latest version 0.2.2

It allows modifying the logits in-place without copying entire data between C and Python:

```py
llm.eval(tokens)

# Use them directly with custom sampling methods
logits = llm.logits

# or modify in-place and call sample()
logits[some_token_id] -= 1
for i in range(len(logits)):
    logits[i] *= 1.1

# sample() method will use the modified logits directly
token = llm.sample()
```

**Comment by m1chae1bx** [comment]
Wow, thank you so much! This is great!

**Comment by domguia** [comment]
`llm.logits` is not available for me. @marella There's a specific way of initializing the llm?
below how I currently initializing llm.
```python
from ctransformers import AutoModelForCausalLM, AutoTokenizer
llm = AutoModelForCausalLM.from_pretrained("TheBloke/LLaMa-7B-GGML", hf=True)
```
Environment:
Colab with Python 3.10
transformers 4.34.1
ctransformers 0.2.27
